docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+		",
+		"rendered": ...
+	}
+}
+```
+
+This triggers a migration. `'footer'` now points to a numeric ID instead of an array. Other widget areas are not affected.
+
+```php
+array(
+	'footer' => 123,
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 4,
+)
+```
+
+Here, `123` is the ID of a `wp_area` post. The `post_content` field of `get_post( 123 )` contains the `'footer'` widget area's block markup.
+
+```html
+<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+<!-- wp:legacy-widget {"identifier":"search-2","instance":{"title":""}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-posts-2","instance":{"title":"","number":5}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-comments-2","instance":{"title":"","number":5}} /-->
+```
+
+## New APIs
+
+### `register_widget_area()`
+
+The `register_widget_area()` function is equivalent to [`register_sidebar()`][register_sidebar] except that it does not support `'class'`, `'before_widget'`, `'after_widget'`, `'before_title'` and `'after_title'`.
+
+By encouraging theme developers to use `register_widget_area()` instead of `register_sidebar()`, we encourage themes to allow users to have full control of their widget areas using the block editor.
+
+```php
+function register_widget_area( array|string $attributes = array() )
+```
+
+#### Parameters
+
+- `$attributes`: _(array|string)_ _(optional)_ Arguments for the widget area being registered.
+  - `'name'`: _(string)_ _(optional)_ Name or title of the widget area. Default `"Widget Area $instance"`.
+  - `'id'`: _(string)_ _(optional)_ Unique identifier of the widget area. Default `"widget-area-$instance"`.
+  - `'description'`: _(string)_ _(optional)_ Description the widget area, displayed in the admin interface.
+
+#### Return
+
+_(string)_ ID added to `$wp_registered_sidebars` global.
+
+### `widget_area()`
+
+The `widget_area()` function is an exact alias of the [`dynamic_sidebar()`][dynamic_sidebar] function, provided so that themes can transition away from thinking about "sidebars".
+
+## Backwards Compatibility
+
+### `wp_get_sidebars_widgets()`
+
+[`wp_get_sidebars_widgets()`][wp_get_sidebars_widgets] can now potentially return an associative array that maps strings to integers. This differs from the current behaviour which is to always return an associative array that maps strings to arrays.
+
+Since `wp_get_sidebars_widgets()` is marked `@private`, no effort is made to ensure backwards compatibility beyond incrementing the `'array_version'` field from `3` to `4`.
```

Comment:
We need to check if any popular plugins are doing something here.

Reply:
Good idea. How do we do this?

Reply:
used to be able to search code found in  plugins in the wp.org repository using wpdirectory.net but it looks like that service is now down.

Reply:
wpdirectory.net is up again now and reports 648 plugins using wp_get_sidebars_widgets().

Although marked as private, I don't think there is any other method for plugin authors to use if they need to get a list of all widgets within a sidebar?

There are 103 plugins using wp_set_sidebar_plugins() also.

Reply:
I've used this function a lot in widget plugins I've built. I think backward compatibility for this function and `wp_set_sidebars_widgets` need to be maintained.

Reply:
> I've used this function a lot in widget plugins I've built. I think backward compatibility for this function and wp_set_sidebars_widgets need to be maintained.

Done in ded575402c55ee46a1b4e48a85cdee1cf0c8b733.